universal-application-tool-0.0.1/app/views/dev/AwsViewStorageStrategy.java

```diff
@@ -0,0 +1,58 @@
+package views.dev;
+
+import static j2html.TagCreator.form;
+import static j2html.TagCreator.input;
+import static j2html.TagCreator.text;
+import static j2html.attributes.Attr.ENCTYPE;
+
+import j2html.TagCreator;
+import j2html.tags.ContainerTag;
+import services.cloud.StorageUploadRequest;
+import services.cloud.aws.SignedS3UploadRequest;
+import views.HtmlBundle;
+import views.ViewUtils;
+
+/** Strategy class for creating a file upload form for AWS. */
+public class AwsViewStorageStrategy implements FileUploadViewStorageStrategy {
+
+  @Override
+  public ContainerTag getFileUploadForm(
+      ViewUtils viewUtils, StorageUploadRequest storageUploadRequest, HtmlBundle bundle) {
+    if (!(storageUploadRequest instanceof SignedS3UploadRequest)) {
+      return null;
+    }
+    SignedS3UploadRequest request = (SignedS3UploadRequest) storageUploadRequest;
+
+    ContainerTag formTag =
+        form()
+            .attr(ENCTYPE, "multipart/form-data")
+            .with(input().withType("input").withName("key").withValue(request.key()))
+            .with(
+                input()
+                    .withType("hidden")
+                    .withName("success_action_redirect")
+                    .withValue(request.successActionRedirect()))
+            .with(
+                input()
+                    .withType("text")
+                    .withName("X-Amz-Credential")
+                    .withValue(request.credential()));
+    if (!request.securityToken().isEmpty()) {
```

Comment:
question (blocking): when would the security token be empty?

Reply:
This is the way the existing code was written, we didn't change this from the existing AWS dev file upload implementation.